src/main/java/io/lbry/browser/reactmodules/DownloadManagerModule.java

```diff
@@ -191,4 +240,24 @@ private String formatBytes(double bytes)
 
         return String.format("%s GB", DECIMAL_FORMAT.format(bytes / (1024.0 * 1024.0 * 1024.0)));
     }
+
+    private static String formatFilename(String filename, int alternateMaxLength) {
+        int maxLength = alternateMaxLength > 0 ? alternateMaxLength : MAX_FILENAME_LENGTH;
+        if (filename.length() < maxLength) {
+            return filename;
+        }
+
+        // Get the extension
+        int dotIndex = filename.lastIndexOf(".");
+        if (dotIndex > -1) {
+            String extension = filename.substring(dotIndex);
+            return String.format("%s...%s", filename.substring(0, maxLength - extension.length() - 4), extension);
+        }
+
+        return String.format("%s...", filename.substring(0, maxLength - 3));
```

Comment:
Why does a a function of this name (seemingly reusable) add trailing periods?
May want to rename the function or add the trailing periods elsewhere.

Reply:
The function is meant to shorten the filename (if longer than a particular length) for the notification content title. I suppose I could rename it to `truncateFilename`.